kie-soup-dataset/kie-soup-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/dialect/OracleDialect.java

```diff
@@ -100,8 +100,12 @@ public String getOffsetLimitSQL(Select select) {
         int offset = select.getOffset();
         int limit = select.getLimit();
         StringBuilder out = new StringBuilder();
-        if (offset > 0) out.append(" OFFSET ").append(offset).append(" ROWS");
-        if (limit >= 0) out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
+        if (offset >= 0) {
+            out.append(" OFFSET ").append(offset).append(" ROWS");
+        }
+        if (limit >= 0) {
+            out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
+        }
```

Comment:
by default, `offset` for OracleDialect is set to `-1`. So this condition is not met at startup and it's only adding `FETCH` to the query.

Maybe we could add a third check here for this particular scenario?

```suggestion
        if (offset == -1 && limit == 0) {
            out.append(" OFFSET 0 ROWS FETCH FIRST 0 ROWS ONLY");
        } else { 
            if (offset >= 0) {
                out.append(" OFFSET ").append(offset).append(" ROWS");
            }
            if (limit >= 0) {
                out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
            }        
        }
```

Reply:
Hmmm, I will use 
```
        if (limit == 0) {
        	return " OFFSET 0 ROWS FETCH FIRST 0 ROWS ONLY";
        } else {
        	StringBuilder out = new StringBuilder();
        	if (offset> 0) out.append(" OFFSET ").append(offset).append(" ROWS");
            if (limit > 0) out.append(" FETCH FIRST ").append(limit).append(" ROWS ONLY");
            return out.toString();
        }
```